libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx

```diff
@@ -193,14 +197,30 @@ const ScopeCardNode = ({ data, targetPosition = Position.Top, sourcePosition = P
     WorkflowService().resubmitWorkflow?.(runInstance?.name ?? '', [id]);
   }, [runInstance, id]);
 
+  const runAfterClick = useCallback(() => {
+    dispatch(changePanelNode(scopeId));
+    dispatch(selectPanelTab(panelTabName));
+  }, [dispatch, scopeId]);
+
+  const shouldDisplayRunAfter = (operation: LogicAppsV2.ActionDefinition): boolean => {
+    if (!operation || !operation.runAfter) {
+      return false;
+    }
+    return Object.keys(operation.runAfter).length > 0;
+  };
+
+  const operationFromWorkflow = getRecordEntry(rootState.workflow.operations, scopeId) as LogicAppsV2.OperationDefinition;
+  const runAfter = isTrigger ? false : shouldDisplayRunAfter(operationFromWorkflow);
```

Comment:
just move the trigger check into the shouldDisplayRunAfter, why keep it seperate?